addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -12,6 +12,14 @@
 class Mailing(models.Model):
     _inherit = 'mailing.mailing'
 
+    @api.model
+    def default_get(self, fields):
+        res = super(Mailing, self).default_get(fields)
+        print(res)
+        if 'mailing_type' in res and res['mailing_type'] == 'sms':
+            res['keep_archives'] = True
```

Comment:
To be complete you should check that either fields is void, either keep_archives is in fields (you may ask only for some specific fields, in which case you don't want to add keep archives in it).